addon/addon.go

```diff
@@ -166,3 +175,94 @@ func NewAddonManager(kubeConfig *rest.Config, agentImage string, agentInstallAll
 
 	return addonMgr, err
 }
+
+func GetMchImage(kubeConfig *rest.Config) (string, error) {
+	kubeClient, err := client.New(kubeConfig, client.Options{})
+	if err != nil {
+		klog.Errorf("Unable to kube client: %v", err)
+
+		return "", err
+	}
+
+	var acm bool
+	acm, err = isACM(kubeClient)
+
+	if err != nil {
+		return "", err
+	}
+
+	// Not ACM
+	if !acm {
+		klog.V(2).Info("ACM not detected")
+
+		return "", nil
+	}
+
+	klog.V(2).Info("ACM detected, wait for MCH config map")
+
+	// Get ACM version
+	acmVersion, found := os.LookupEnv(OperatorVersionEnv)
+	if !found {
+		err = fmt.Errorf("%v env var not found", OperatorVersionEnv)
+		klog.Error(err)
+
+		return "", err
+	}
+
+	mchImageCMkey := types.NamespacedName{
+		Name:      "mch-image-manifest-" + acmVersion,
+		Namespace: "open-cluster-management",
+	}
+
+	klog.Infof("Get MCH image config map: %v/%v", mchImageCMkey.Namespace, mchImageCMkey.Name)
+
+	mchImageCM := &corev1.ConfigMap{}
+
+	for count := 0; count < 60; count++ {
+		klog.V(2).Infof("Waiting for MCH config map, count: %v", count)
```

Comment:
we can explicitly show the waiting message for user awareness. So V(2) is not necssary.